clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -61,22 +67,32 @@ public Builder(String groupId, List<TopicPartition> partitions) {
 
             this.data = new OffsetFetchRequestData()
                             .setGroupId(groupId)
+                            .setRequireStable(requireStable)
                             .setTopics(topics);
         }
 
-        public static Builder allTopicPartitions(String groupId) {
-            return new Builder(groupId, null);
-        }
-
-        public boolean isAllTopicPartitions() {
+        boolean isAllTopicPartitions() {
             return this.data.topics() == ALL_TOPIC_PARTITIONS;
         }
 
         @Override
         public OffsetFetchRequest build(short version) {
-            if (isAllTopicPartitions() && version < 2)
+            if (isAllTopicPartitions() && version < 2) {
                 throw new UnsupportedVersionException("The broker only supports OffsetFetchRequest " +
-                        "v" + version + ", but we need v2 or newer to request all topic partitions.");
+                    "v" + version + ", but we need v2 or newer to request all topic partitions.");
+            }
+
+            if (data.requireStable() && version < 7) {
+                log.warn("Fallback the requireStable flag to false as broker " +
```

Comment:
Like you said, that would depend on how frequent we do offset fetch, maybe @guozhangwang knows any other occurrence than rebalance?

Reply:
In streams for example we call `consumer#committed` whenever registering a changelog for restoration, other than that it may be called fairly infrequently. But I'd suggest we use INFO not WARN since for some other cases where we downgrade based on APIVersion I remember we used INFO too.

Reply:
Sounds good!